src/cairoUtilFuncGen/memory/arrayLiteral.ts

```diff
@@ -5,39 +5,68 @@ import {
   FunctionCall,
   generalizeType,
   getNodeType,
+  IntType,
+  Literal,
+  LiteralKind,
   TupleExpression,
   TypeNode,
 } from 'solc-typed-ast';
 import { printNode } from '../../utils/astPrinter';
 import { CairoType } from '../../utils/cairoTypeSystem';
-import { NotSupportedYetError } from '../../utils/errors';
+import { cloneASTNode } from '../../utils/cloning';
 import { createCairoFunctionStub, createCallToFunction } from '../../utils/functionGeneration';
+import { createArrayTypeName, createNumberLiteral } from '../../utils/nodeTemplates';
 import { notNull } from '../../utils/typeConstructs';
 import { mapRange, narrowBigIntSafe, typeNameFromTypeNode } from '../../utils/utils';
 import { uint256 } from '../../warplib/utils';
 import { add, locationIfComplexType, StringIndexedFuncGen } from '../base';
 
 /*
-  Converts [a,b,c] into WM0_arr(a,b,c), which allocates new space in warp_memory
+  Converts [a,b,c] and "abc" into WM0_arr(a,b,c), which allocates new space in warp_memory
   and assigns the given values into that space, returning the location of the
   start of the array
 */
 export class MemoryArrayLiteralGen extends StringIndexedFuncGen {
-  gen(node: TupleExpression): FunctionCall {
+  stringGen(node: Literal): FunctionCall {
+    // Encode the literal to the uint-8 byte representation
+    assert(node.kind === LiteralKind.String);
```

Comment:
What about LiteralKind.UnicodeString (or even hexString, is that relevant here?)